clients/drcachesim/scheduler/scheduler.h

```diff
@@ -1485,6 +1504,21 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
     memtrace_stream_t *
     get_input_stream(output_ordinal_t output);
 
+    // Returns the possibly adjusted (for queued records) record ordinal for the current
+    // input stream interface for the 'output_ordinal'-th output stream.
+    uint64_t
+    get_input_record_ordinal(output_ordinal_t output);
+
+    // Returns the possibly adjusted (for queued records) first timestamp for the current
+    // input stream interface for the 'output_ordinal'-th output stream.
+    uint64_t
+    get_input_first_timestamp(output_ordinal_t output);
+
+    // Returns the possibly adjusted (for queued records) last timestamp for the current
+    // input stream interface for the 'output_ordinal'-th output stream.
+    uint64_t
+    get_input_last_timestamp(output_ordinal_t output);
```

Comment:
Could we add some checks in  scheduler_unit_tests.cpp for these new apis?

Reply:
These are not public so the unit tests can't directly call these (without some new subclass).  The tests of the public stream APIs do exercise these as they directly call them for analyzer parallel and serial modes.

Reply:
These were the main things I had to work on to get all the tests to pass: multiple tests fail if something is off here, from unit tests to larger tests.